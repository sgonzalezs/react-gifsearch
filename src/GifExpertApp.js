import React, {useState} from 'react';
import {AddCategory} from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp=({defaultCategories=[]})=>{

    // const [categories, setCategories] = useState(['space']);
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd=()=>{
    //     /**
    //      * 1 - Obtenemos las categorias del primer estado con ...categories
    //      * y agregamos la nueva categoria (Space)
    //      */

    //     // const newCategories=[...categories, 'Space'];
    //     // setCategories(newCategories);

    //     /**
    //      * 2 - Obtenemos el estado anterior por medio de un callback
    //      * y agregamos la nueva categoria (Space)
    //      */
    //     setCategories((catgs)=>{
    //         return [...catgs, 'space'];
    //     });
    // }

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
                {
                    categories.map((category)=>{
                        return(
                            <GifGrid 
                                key={category} 
                                category={category} 
                            />
                        )
                    })
                }
        </>
    );
}

export default GifExpertApp;