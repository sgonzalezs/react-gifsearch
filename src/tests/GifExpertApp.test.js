import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Tests File GifExpertApp', () => {

    test('Show component <GifExpertApp/>', () => {

        const wrapper=shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Must show a list of categories', () => {
        
        const categories=['Space', 'Andromada'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    
});
