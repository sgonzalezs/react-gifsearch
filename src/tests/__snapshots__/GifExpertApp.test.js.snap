// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests File GifExpertApp Must show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Space"
    key="Space"
  />
  <GifGrid
    category="Andromada"
    key="Andromada"
  />
</Fragment>
`;

exports[`Tests File GifExpertApp Show component <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
